source/beamline/SGM/energy/SGMUndulatorSupport.h

```diff
@@ -95,7 +97,7 @@ inline static double optimizedUndulatorPosition(double energy,
 				  SGMUndulatorSupport::UndulatorHarmonic undulatorHarmonic,
 				  double undulatorOffset)
 {
-	return (-1/0.14295709668) * log( (1/36.00511212946)*((1737.41045746644/(energy/int(undulatorHarmonic))) -1)) + undulatorOffset;
+	return qMin(qMax((-1/0.14295709668) * log( (1/36.00511212946)*((1737.41045746644/(energy/int(undulatorHarmonic))) -1)) + undulatorOffset, UNDULATOR_MIN_POSITION), UNDULATOR_MAX_POSITION);
```

Comment:
You should use qBound().  Then this will look like,

```c++
double equationResult = < equation >;
return qBound(UNDULATOR_MIN_POSITION, equationResult, UNDULATOR_MAX_POSITION);
```